### DISCUSSION TITLE
Dyson Sphere

### DISCUSSION THREAD

**Discussion by JustPyrrha** [TeamGalacticraft/Galacticraft#174] [Feature Requests]
Info [Wikipedia](https://en.wikipedia.org/wiki/Dyson_sphere)

**Comment by craftablescience** [comment]
Few questions for clarity:

1. How would power be transmitted from the sphere to Earth / other planets / space stations? Would there have to be some sort of large, multiblock structure or a simple block on the receiving end?
2. Would there be any visual markings on the Sun?
3. The Sun is large, and a Dyson Sphere would likely take a LOT of resources. How would it be craftable? Would it require some sort of autonomous launch? We should probably add a satellite delivery system before this is implemented.
4. How would this be practical? We can already generate cheap energy from solar panel arrays. Currently, no Galacticraft spaceship is powered in any way by electricity. For interstellar warp drive travel, a Dyson Sphere might help with power needs, but that type of travel does not exist right now in-game.

These questions do not have to be answered, I just think we should consider them before implementing this technology.

**Comment by Archilyte** [comment]
[Folks at the other mod](https://github.com/zmaster587/AdvancedRocketry/issues/1714) had a big discussion about this. What I can get form it is that a Dyson Sphere in Minecraft wouldn't be worth the cost. The idea of a Dyson Sphere has trillions of people's needs in mind, and I have yet to see a server with at least a million players. And most modpacks already have a progression towards a creative power source, which would make the Dyson Sphere useless, and even then you barely use the Creative energy source.

I would rather not bother with a Dyson Sphere.

**Comment by TrueCP6** [comment]
If you created the need by requiring enormous amounts of energy for interstellar/intergalactic travel via something like a light sail you could justify the need for a Dyson Cluster/Ring.

**Comment by Archilyte** [comment]
That does remind me of an upcoming game coming tomorrow called [Dyson Sphere Program](https://store.steampowered.com/app/1366540/Dyson_Sphere_Program/). Maybe we could take inspiration?

**Comment by voidsong-dragonfly** [comment]
As well, light sails, while possibly being able to _use_ a dyson swarm, would use a small potion of one at best - not anywhere close to the full thing. You'd be better off with just launching large numbers of solar satellites somehow rather than making (or trying to make) a full dyson swarm. This is both easier to automate on the player, more realistic for the player to achieve in any reasonable timescale. Microwave satellites/receiver pairs is your best bet to mimic this in MC. Even interstellar drives, whether they're below C or very very close to it, do not need so much power as to cut into the 384 yottawatts of power the sun generates, even if it were collected only partially efficiently by a dyson swarm.

Archilyte is absolutely right, the only sinks in modded that need power in close-to-creative levels are "fill up the stupid ball" and whatever quests modpacks create to take huge amounts of power - and those will scale with generators. With things like the rainbow generator and creative power sources, the sheer complexity to launch and position a dyson swarm would quickly become so prohibitive as to be useless were you to try and use the power for actual FE/etc.

Which leaves us with no need for a dyson swarm in any real capacity.
